### PR TITLE
To do form component

### PR DESCRIPTION
**Pull request by nikitha2000**
To do form compoent

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -26,7 +27,70 @@ button{
...
+  padding: 20px;
+  border-radius: 8px;
+  box-shadow: 0 4px 8px 4px rgba(0, 0, 0, 0.1);
+  width: 300px;
+  margin:0 auto 10px auto;
+  align-self: center;
+  margin-top: 10px;
+}
+
+.TodoInput{
```
Comment (vonnue-nidheesh) [inline src/App.css]: The CSS classname must be Hyphen Delimited Strings.

Refer: https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

File: src/App.css [hunk 2]
```diff
@@ -26,7 +27,70 @@ button{
...
+  width: calc(100% - 100px);
+  font-size: 1em;
+  margin-bottom: 10px;
+  border: 1px solid #ccc;
+  border-radius: 5px;
+  box-sizing: border-box;
+  margin-left:5px
+}
+ 
+.userStatus select{
```
Comment (vonnue-nidheesh) [inline src/App.css]: Same here

File: src/App.css [hunk 3]
```diff
@@ -26,7 +27,70 @@ button{
...
+  grid-template-columns: 1fr auto;
+  gap: 10px;
+  padding: 10px;
+  margin-bottom: 8px;
+  background-color: #f9f9f9;
+  border:1px solid #ddd;
+  border-radius: 5px;
+  font-size: 1.1em; 
+}
+ul{
```
Comment (vonnue-nidheesh) [inline src/App.css]: New line between `ul` and `{`

File: src/App.js [hunk 4]
```diff
@@ -1,17 +1,50 @@
 import React, { useState } from "react"
 import './App.css';
 import Button from './component/Button';
+import ToDoForm from "./component/TodoForm";
 
+function App() {
 
 
+  const [clickedButton, setClickedButton] = useState(false)
+  const[todos, setTodo] = useState([])
```
Comment (vonnue-nidheesh) [inline src/App.js]: In this file, some lines has semicolon at the end and some don't. Why like that?

File: src/App.js [hunk 5]
```diff
@@ -1,17 +1,50 @@
...
+ 
 
+  const handleClick =  (buttonId) => {
```
Comment (vonnue-nidheesh) [inline src/App.js]: `buttonID`

Comment (vonnue-nidheesh) [inline src/App.js]: The function name is too generic.
rename something like 'layoutToggleButtonClick'

File: src/App.js (hunk 4 above)
Comment (vonnue-nidheesh) [inline src/App.js]: setTodos

Comment (nikitha2000) [reply src/App.js]: I forgot to put it in some places.

File: src/App.js [hunk 6]
```diff
@@ -1,17 +1,50 @@
...
+    return (
+      <div className="App">
+        <div className='app-container'>
+          <h1>TODO App</h1>
+            <div className='button-container'>
+               <Button color="#8888b9" label="Todo" onClick={() => {handleClick("button1")}}  isClicked={clickedButton === "button1"}  />
+               <Button color="#8888b9" label="Todo list" onClick={() => {handleClick("button2")}}  isClicked={clickedButton === "button2"} />
+            </div>
+           
+            {clickedButton=='button1' && (
```
Comment (vonnue-nidheesh) [inline src/App.js]: `clickedButton=='button1'` this is reused in some places. So, declare a variable for this and reuse it everywhere.

eg:
```
const isTodoButtonClicked = clickedButton=='button1';

{isTodoButtonClicked && <ToDoForm onAddTodo={addTodo} />}
```

File: src/App.js [hunk 7]
```diff
@@ -1,17 +1,50 @@
...
 
-function App() {
+
+   
+    return (
+      <div className="App">
+        <div className='app-container'>
+          <h1>TODO App</h1>
+            <div className='button-container'>
+               <Button color="#8888b9" label="Todo" onClick={() => {handleClick("button1")}}  isClicked={clickedButton === "button1"}  />
```
Comment (vonnue-nidheesh) [inline src/App.js]: rename `button1` to `todoButton`'

File: src/App.js [hunk 8]
```diff
@@ -1,17 +1,50 @@
...
+               <Button color="#8888b9" label="Todo list" onClick={() => {handleClick("button2")}}  isClicked={clickedButton === "button2"} />
```
Comment (vonnue-nidheesh) [inline src/App.js]: rename `button2` to `todoListButton`

File: src/App.js [hunk 9]
```diff
@@ -1,17 +1,50 @@
...
+}
 
-function App() {
+
+   
+    return (
+      <div className="App">
+        <div className='app-container'>
+          <h1>TODO App</h1>
+            <div className='button-container'>
```
Comment (vonnue-nidheesh) [inline src/App.js]: Rename CSS class `button-container`  with `layout-buttons-container`

File: src/App.js [hunk 10]
```diff
@@ -1,17 +1,50 @@
...
+               <Button color="#8888b9" label="Todo" onClick={() => {handleClick("button1")}}  isClicked={clickedButton === "button1"}  />
+               <Button color="#8888b9" label="Todo list" onClick={() => {handleClick("button2")}}  isClicked={clickedButton === "button2"} />
+            </div>
+           
+            {clickedButton=='button1' && (
+              <ToDoForm onAddTodo={addTodo} />
+            )}
+
+
```
Comment (vonnue-nidheesh) [inline src/App.js]: Remove unwanted empty lines

File: src/App.js [hunk 11]
```diff
@@ -1,17 +1,50 @@
...
 
 export default App;
 
+
                  
     
-     
\ No newline at end of file
+
+
```
Comment (vonnue-nidheesh) [inline src/App.js]: Remove unwanted empty lines

File: src/component/TodoForm.jsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
+    import React, { useState } from 'react'
+    import Button from './Button'
+    
+    const ToDoForm = ({onAddTodo}) => {
+
+
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: Remove one empty line from here

File: src/component/TodoForm.jsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        const [TaskName, setTodoText] = useState('');
+        const [TaskStatus, setTodoStatus]=useState('todo');
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: Variable name must be in camelCase

taskName, taskStatus

File: src/component/TodoForm.jsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+        return ( 
+        <form className='ToDoForm' onSubmit={handleSubmit}>  
+             <label htmlFor='taskname'>Enter to do:</label>        
+             <input className='TodoInput' type='text' placeholder='Enter to do' value={TaskName} onChange={handleInputChange} maxLength={maxlength}></input>
+              <div className="userStatus">
+                  <label htmlFor='userstatus'>Todo Status:</label>
+                  <select value={TaskStatus} onChange={handleStatusChange}>
+                     <option value="Todo">Todo</option>
+                     <option value="In-Progress">In-Progress</option>
+                     <option value="Completed">Completed</option>
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: create a constant variable for these items and map it here.

eg: 
```

const taskStatuses = ['Todo',  'In-Progress', 'Completed' ];

 <select value={TaskStatus} onChange={handleStatusChange}>
        { taskStatuses.map(....)}
 </select>
```

File: src/component/TodoForm.jsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+                  </select>
+              </div>
+            <Button color='#8888b9' label='Create To Do' onClick={handleSubmit} isClicked={false} isDisabled={isButtonDisabled} />
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: There is no need to pass the isClicked prop from here.

File: src/component/TodoForm.jsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+        </form>
+        )
+    }
+
+export default ToDoForm
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: semicolon required at the end

File: src/component/TodoForm.jsx [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+        
+        const handleInputChange = (e) =>{
+           setTodoText(e.target.value)
+        }
+
+        const handleStatusChange = (e) => {
+          setTodoStatus (e.target.value)
+        }
+
+        const isButtonDisabled = TaskName.trim() === '';
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: There is no need of a new variable for this we can directly use it like this 'isDisabled={!taskName.trim()}'

File: src/App.js (hunk 5 above)
Comment (nikitha2000) [reply src/App.js]: ok

File: src/component/Button.jsx [hunk 18]
```diff
@@ -0,0 +1,22 @@
+import React from "react"
+
+const Button = ({ color='#8888b9', label, onClick, isClicked=false, isDisabled=false}) => {
```
Comment (vonnue-nidheesh) [inline src/component/Button.jsx]: It is better to keep optional props at last.
`const Button = ({ label, onClick, color='#8888b9', isClicked=false, isDisabled=false}) => {`

File: src/App.js [hunk 19]
```diff
@@ -1,17 +1,39 @@
...
-
-
+import ToDoForm from "./component/TodoForm";
 
 function App() {
-
-}
-
+  const [clickedButton, setClickedButton] = useState(false);
+  const[todos, setTodo] = useState([]);
```
Comment (vonnue-nidheesh) [inline src/App.js]: Format file. You can install VS code extension Prettier - Code formatted for format code.

`const [todos, setTodo] = useState([]);`

File: src/App.css [hunk 20]
```diff
@@ -26,7 +27,66 @@ button{
...
+  padding: 20px;
+  border-radius: 8px;
+  box-shadow: 0 4px 8px 4px rgba(0, 0, 0, 0.1);
+  width: 300px;
+  margin:0 auto 10px auto;
+  align-self: center;
+  margin-top: 10px;
+}
+
+.todo-input{
```
Comment (vonnue-nidheesh) [inline src/App.css]: `.todo-input {`

File: src/component/TodoForm.jsx [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+                  <label htmlFor='userstatus'>Todo Status:</label>
+                  <select value={taskStatus} onChange={handleStatusChange}>
+                  {taskStatuses.map((status) => (
+                          <option key={status} value={status}>
+                              {status}
+                          </option>
+                   ))}
+                  </select>
+              </div>
+            <Button color='#8888b9' label='Create To Do' onClick={handleSubmit} isDisabled={taskName.trim() === ''} />
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: `isDisabled={!taskName.trim()}`

File: src/component/TodoForm.jsx [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+        </form>
+        );
+    };
+export default ToDoForm;
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx:48]: Empty lines are needed before and after.

File: src/component/TodoForm.jsx [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+                  <label htmlFor='userstatus'>Todo Status:</label>
+                  <select value={taskStatus} onChange={handleStatusChange}>
+                  {taskStatuses.map((status) => (
+                          <option key={status} value={status}>
+                              {status}
+                          </option>
+                   ))}
+                  </select>
+              </div>
+            <Button color='#8888b9' label='Create To Do' onClick={handleSubmit} isDisabled={!taskName.trim() === ''} />
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: `isDisabled={!taskName.trim()}`

File: src/App.js [hunk 24]
```diff
@@ -3,38 +3,37 @@ import './App.css';
...
+  const isTodoListButtonClicked = clickedButton == 'todoListButton';
+  
+ return (
       <div className="App">
         <div className='app-container'>
           <h1>TODO App</h1>
-            <div className='button-container'>
-               <Button color="#8888b9" label="Todo" onClick={() => {handleClick("button1")}}  isClicked={clickedButton === "button1"}  />
-               <Button color="#8888b9" label="Todo list" onClick={() => {handleClick("button2")}}  isClicked={clickedButton === "button2"} />
+            <div className='layout-button-container'>
```
Comment (vonnue-nidheesh) [inline src/App.js]: `layout-buttons-container`

File: src/component/TodoForm.jsx [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+    
+    const ToDoForm = ({onAddTodo}) => {
+        
+        const [taskName, setTodoText] = useState('');
+        const [taskStatus, setTodoStatus]=useState('todo');
+        const maxlength = 25;
+
+         const handleSubmit = (e) => {
+            e.preventDefault();
+            if(TaskName.trim()) {
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: taskName

File: src/component/TodoForm.jsx [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+                onAddTodo({text:taskName, status:taskStatus});
+                setTodoText('');
+                setTodoStatus('Todo');
+            };
+         };
+        
+        const handleInputChange = (e) =>{
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: `handleTodoInputChange`

File: src/component/TodoForm.jsx [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+           setTodoText(e.target.value);
```
Comment (vonnue-nidheesh) [inline src/component/TodoForm.jsx]: It is better to trim the value here. and avoid trim() in other places.

**Review by vonnue-nidheesh — APPROVED** [review]

**Pull request merged**